pkg/helm/helm.go

```diff
@@ -0,0 +1,285 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helm
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	metallbv1beta1 "github.com/metallb/metallb-operator/api/v1beta1"
+	"github.com/pkg/errors"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/cli/values"
+	"helm.sh/helm/v3/pkg/getter"
+	apiext "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	helmReleaseName = "metallb"
+	bgpFrr          = "frr"
+)
+
+type Parser interface {
+	GetManifests(config *metallbv1beta1.MetalLB, client client.Client,
+		isOpenShift bool) ([]*unstructured.Unstructured, error)
+	GetChart() *chart.Chart
+	GetClient() *action.Install
+}
+
+type helm struct {
+	client      *action.Install
+	envSettings *cli.EnvSettings
+	chart       *chart.Chart
+}
+
+func NewParser(helmChartPath string, helmEnvSettings *cli.EnvSettings) (Parser, error) {
+	helm := &helm{}
+	helm.envSettings = helmEnvSettings
+	helm.client = action.NewInstall(new(action.Configuration))
+	helm.client.ReleaseName = helmReleaseName
+	helm.client.Namespace = helmEnvSettings.Namespace()
+	helm.client.DryRun = true
+	helm.client.Replace = true
+	helm.client.ClientOnly = true
+	chartPath, err := helm.client.ChartPathOptions.LocateChart(helmChartPath, helmEnvSettings)
+	if err != nil {
+		return nil, err
+	}
+	helm.chart, err = loader.Load(chartPath)
+	if err != nil {
+		return nil, err
+	}
+	return helm, nil
+}
+
+func (h *helm) GetChart() *chart.Chart {
+	return h.chart
+}
+
+func (h *helm) GetClient() *action.Install {
+	return h.client
+}
+
+func (h *helm) GetManifests(config *metallbv1beta1.MetalLB, client client.Client,
+	isOpenShift bool) ([]*unstructured.Unstructured, error) {
+	chartValueOpts := &values.Options{}
+	chartValues, err := chartValueOpts.MergeValues(getter.All(h.envSettings))
+	if err != nil {
+		return nil, err
+	}
+	err = patchPrometheusValues(chartValues, client)
```

Comment:
I'd move this config -> values translation logic in a separate file (still in this pkg)